#!/usr/bin/env python

PACKAGE = 'kxr_controller'

from dynamic_reconfigure.parameter_generator_catkin import int_t
from dynamic_reconfigure.parameter_generator_catkin import double_t
from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator


gen = ParameterGenerator()
# The value 5 for frame_count was experimentally confirmed to enable smooth interpolation
gen.add("frame_count", int_t, 0, "Frame Count", 50, 1, 255)
gen.add("wheel_frame_count", int_t, 0, "Frame Count for Wheel", 1, 1, 255)
gen.add("temperature_limit", double_t, 0, "Temperature limit in celsius", 80, 0, 120)
gen.add("current_limit", double_t, 0, "Current limit in Amp", 4.0, 0.0, 6.0)

exit(gen.generate(PACKAGE, PACKAGE, "KXRParameters"))
